settings/settings-module.php

```diff
@@ -247,14 +245,14 @@ public function get_form() {
 	}
 
 	/**
-	 * Allows child classes to validate their options before saving.
+	 * Allows child classes to prepare the received data before saving.
 	 *
-	 * @since 1.8
+	 * @since 3.7
 	 *
-	 * @param array $options Unsanitized options to save.
-	 * @return array Options
+	 * @param array $options Raw values to save.
+	 * @return array
 	 */
-	protected function update( $options ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	protected function prepare_raw_data( array $options ): array { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 		return array(); // It's responsibility of the child class to decide what is saved.
```

Comment:
`update()` was responsible for the validation. It's not the case anymore for the new method, so I would expect that it returns unchanged options. Only options that need preparation to change the format would need to override this method. Am I wrong?

Reply:
This method has two jobs:
1. Format the data the way the option classes need it.
2. Limit to options handled by the current settings class.

This means all settings classes need to extend this method. And in the spirit of the 2nd point, for me it makes sense to return an empty array instead of all the data.
Does it make sense to you?